apps/deposit-pool/src/hooks/useGetHookInfo.ts

```diff
@@ -0,0 +1,194 @@
+      if (!pool || !context || !context.account || !cowShedProxy)
+        throw new Error("Missing context");
+      const poolState = minimalPoolToPoolState(pool);
+
+      const referenceTokenDecimals = pool.allTokens.find(
+        (token) =>
+          token.address.toLowerCase() ===
+          params.referenceTokenAddress.toLowerCase()
+      )?.decimals;
+
+      if (!referenceTokenDecimals) throw new Error("Invalid reference token");
+
+      const addLiquidityInput: AddLiquidityProportionalInput = {
+        chainId: context.chainId,
+        rpcUrl: RPC_URL_MAPPING[context.chainId],
+        referenceAmount: {
+          rawAmount: parseUnits(
+            params.amounts[
+              params.referenceTokenAddress.toLowerCase()
+            ].toString(),
+            referenceTokenDecimals
+          ),
+          decimals: referenceTokenDecimals,
+          address: params.referenceTokenAddress as Address,
+        },
+        kind: AddLiquidityKind.Proportional,
+      };
+      const query = await addLiquidity.query(addLiquidityInput, poolState);
+
+      const depositArg: BalancerDepositArgs = {
+        type: TRANSACTION_TYPES.BALANCER_DEPOSIT,
+        query,
+        chainId: context.chainId,
+        sender: cowShedProxy,
+        recipient: context.account,
+      };
+
+      const transferFromUserToProxyTxs = query.amountsIn.map((tokenAmount) => {
+        return {
+          type: TRANSACTION_TYPES.ERC20_TRANSFER_FROM,
+          token: tokenAmount.token.address,
+          from: context.account,
+          to: cowShedProxy,
+          amount: tokenAmount.amount,
+          symbol: tokenAmount.token.symbol,
+        } as ERC20TransferFromArgs;
+      });
+
+      const approveArgs = pool.allTokens.map((token) => {
+        return {
+          type: TRANSACTION_TYPES.ERC20_APPROVE,
+          token: token.address as Address,
+          spender: pool.address,
+          amount: MAX_UINT256,
+        } as ERC20ApproveArgs;
+      });
+
+      return Promise.all(
+        [...transferFromUserToProxyTxs, ...approveArgs, depositArg].map((arg) =>
+          TransactionFactory.createRawTx(arg.type, arg)
+        )
+      );
+    },
+    [context, cowShedProxy, pool]
+  );
+
+  const getWeirollTransferFromProxyToUserTxs = useCallback(() => {
+    if (!pool || !context?.account || !cowShedProxy)
+      throw new Error("Missing context");
+
+    const weirollTransferFromProxyArgs = pool.allTokens.map((token) => {
+      const tokenAddress = token.address.toLowerCase();
+      return {
+        type: TRANSACTION_TYPES.ERC20_TRANSFER_FROM_ALL_WEIROLL,
+        token: tokenAddress as Address,
+        from: cowShedProxy,
+        to: context.account,
+        symbol: token.symbol,
+      } as ERC20TransferFromAllWeirollArgs;
+    });
+
+    return Promise.all(
+      weirollTransferFromProxyArgs.map((arg) => {
+        return TransactionFactory.createRawTx(arg.type, arg);
+      })
+    );
+  }, [context?.account, cowShedProxy, pool]);
```

Comment:
I know it is more a question of opinion, but don't you think it would be more straightforward to just create the Tx, instead of creating the args and then the Tx?

```
const getWeirollTransferFromProxyToUserTxs = useCallback(() => {
    if (!pool || !context?.account || !cowShedProxy)
      throw new Error("Missing context");

    return Promise.all(
      pool.allTokens.map((token) => {
        return TransactionFactory.createRawTx(
          TRANSACTION_TYPES.ERC20_TRANSFER_FROM_ALL_WEIROLL,
          {
            type: TRANSACTION_TYPES.ERC20_TRANSFER_FROM_ALL_WEIROLL,
            token: token.address.toLowerCase() as Address,
            from: cowShedProxy,
            to: context.account,
            symbol: token.symbol,
          } as ERC20TransferFromAllWeirollArgs,
        );
      }),
    );
  }, [context?.account, cowShedProxy, pool]);
```